pkg/docker/config/config.go

```diff
@@ -136,7 +136,15 @@ func getAuthFilePaths(sys *types.SystemContext) []authPath {
 		// Logging the error as a warning instead and moving on to pulling the image
 		logrus.Warnf("%v: Trying to pull image in the event that it is a public image.", err)
 	}
+
+	var xdgCfgHomePath string
+	if xdgCfgHome := os.Getenv("XDG_CONFIG_HOME"); xdgCfgHome != "" {
+		xdgCfgHomePath = filepath.Join(xdgCfgHome, xdgRuntimeDirPath)
+	} else {
+		xdgCfgHomePath = filepath.Join(homedir.Get(), nonLinuxAuthFilePath)
```

Comment:
Using a “`nonLinux`” path on Linux is also confusing — and on non-Linux that path was already added.

(In this context, it’s not a “non-linux” code path; it’s a “following the XDG base dir spec, XDG_CONFIG_HOME is not set so use the spec-defined default” path; i.e. the if/else should conceptually only get the $HOME`/.config`)